libvmi/arch/amd64.c

```diff
@@ -137,20 +135,35 @@ status_t get_pte_ia32e (vmi_instance_t vmi,
 }
 
 static inline
-uint64_t get_paddr_ia32e (addr_t vaddr, uint64_t pte)
+uint64_t get_paddr_ia32e (vmi_instance_t vmi, addr_t vaddr, uint64_t pte)
```

Comment:
I would prefer if the vmi instance was not passed to these functions and instead a `max_bits` uint was passed in. It would be 51 by default but could be specified to be 44 where appropriate. Right now you specify 44 for all Windows where it sounds like that's only correct for some newer versions. Should that be more fine grained so it only applies where needed?

Reply:
I would still like to avoid having to pass the `vmi` instance to these functions, so please adjust the prototype of these functions to take the number as input instead of doing a pointer deref and and if/else.

Reply:
I already have, please see my latest force push.